core/services/remote_access_services.py

```diff
@@ -122,7 +124,7 @@ def store_trained_classifier_model(job_result_dict):
         raise Exception('job_result_dict must contain \'classifier_data\'.')
 
     payload = {
-        'message': job_result_dict,
+        'message': json.loads(json.dumps(job_result_dict)),
```

Comment:
This hack is needed, without it signature generated on Oppia and Oppia-ml are different.

Reply:
I thought it might be an issue with the keys of nested dict but json.dumps(sort_keys=True) takes care of that as well. @seanlip you might want to see this....

Reply:
> I thought it might be an issue with the keys of nested dict but json.dumps(sort_keys=True) takes care of that as well. @seanlip you might want to see this....

That's what I, too, observed.

Reply:
Finally removed the hack (thanks @seanlip). The problem was that classifier_data returned by classifier contained `str` literals wherever it used strings (keys, values anywhere) but when you dump it and load it again these same literals are stored as `unicode` literals. So when you transfer data to Oppia across network, what Oppia gets is `unicode` and therefore generates signature on those values whereas Oppia-ml generates them using `str` values. Therefore signature are different on both side.

Also @seanlip said that we should be using unicode everywhere in Oppia-ml. Therefore rather than a converter function (which converts `str` to `unicode` in entire object) I've added checker function (which checks that all key / values are `unicode` otherwise raises exceptions.)